src/backend/SecurityContext.c

```diff
@@ -433,8 +433,8 @@ static int BSL_ExecBCBSource(BSL_SecCtx_Execute_f sec_context_fn, BSL_LibCtx_t *
     }
 
     // Over-allocate size for the ASB BTSD in the block
-    const size_t est_btsd_size = 500 + ((n_params + n_results) * 100);
-    if (BSL_BundleCtx_ReallocBTSD(bundle, sec_blk.block_num, est_btsd_size) != BSL_SUCCESS)
+    const size_t est_asb_bytelen = BSL_AbsSecBlock_Sizeof() + ((n_params + n_results + 1) * BSL_SecParam_Sizeof());
```

Comment:
If this is treating C structure size as encoded size, that will not work in a reliable (architecture- or ABI-independent) way. Is there a reason why an estimate is needed beforehand? The QCBOR library provides a way to get the true encoded size (without actually encoding) but at the cost of duplicate processing.